pkgs/development/python-modules/pgsanity/default.nix

```diff
@@ -0,0 +1,30 @@
+{ stdenv
+, fetchFromGitHub
+, buildPythonPackage
+, postgresql }:
+
+buildPythonPackage rec {
+  pname = "pgsanity";
+  version = "0.2.9";
+  rev = "3bd391be1c5f0e2ce041652bdaf1d6b54424c6d8";
+
+  src = fetchFromGitHub {
+    owner = "markdrago";
+    repo = "pgsanity";
+    sha256 = "1imdgzh0nyq43w41wr62wx25wg45fjz3azwybsla81lxxp4rlvjg";
+    inherit rev;
+  };
+
+  doCheck = false;
```

Comment:
What are the reasons to skipi the `checkPhase`? If that's necessary e.g. for impure tests, please leave a comment here as well :)

Reply:
@Ma27 That was fully cargo-cult of me. I did not study that `check phase` variable and assumed that it would be okay to do that, as most python module derivations have it disabled -
> grep -i 'doCheck' pkgs/development/python-modules/\*/default.nix | grep -c false
881
> grep -i 'doCheck' pkgs/development/python-modules/\*/default.nix | grep -c -v false
93

Let me enable it and report back.

Reply:
> it would be okay to do that, as most python module derivations have it disabled

That's not entirely true: `ls -l | wc -l` is 2235, so in fact the majority has the check phase enabled (`doCheck` is `true` by default when using `buildPythonPackage`, so I'm afraid that the `grep -v` check was a false-positive).

Let's see what happens if it's enabled and if there are issues that should be easy to fix (such as a missing `pytest` which is needed to run the tests).

As soon as the tests have more complex issues (usually related to networking or IO operations that break in a sandboxed build environment) you may want to try skipping some of the tests. But in the end it's absolutely ok to disable a check phase with too many errors, but please leave a comment then :)

Reply:
@Ma27 Ah, I had [read this](https://nixos.org/nixpkgs/manual/#ssec-check-phase) in the manual and thought that it is disabled by default -

**doCheck**
```
Controls whether the check phase is executed. By default it is skipped, but if doCheck is set to true, the check phase is usually executed.
```
I suppose, then that this behaviour is overridden by the `buildPythonPackage` function as mentioned in [the Reference here](https://nixos.org/nixpkgs/manual/#interpreters).

I have enabled checked and the package builds normally as well with `nox-review`.